extensions/rich_text_components/Math/directives/oppia-noninteractive-math.directive.ts

```diff
@@ -41,6 +49,52 @@ angular.module('oppia').directive('oppiaNoninteractiveMath', [
             $attrs.mathContentWithValue);
           if (mathExpressionContent.hasOwnProperty('raw_latex')) {
             ctrl.rawLatex = mathExpressionContent.raw_latex;
+            ctrl.svgFilename = mathExpressionContent.svg_filename;
+            var dimensions = ImagePreloaderService.getDimensionsOfMathSvgs(
+              ctrl.svgFilename);
+            ctrl.imageContainerStyle = {
+              height: dimensions.height + 'ex',
+              width: dimensions.width + 'ex',
+              'vertical-align': '-' + dimensions.verticalPadding + 'ex'
+            };
+          }
+          // If viewing a concept card in the exploration player, don't use the
+          // preloader service. Since, in that service, the image file names are
+          // extracted from the state contents to be preloaded, but when
+          // viewing a concept, the names are not available until the link is
+          // clicked, at which point an API call is done to get the skill
+          // details. So, the image file name will not be available to the
+          // preloader service beforehand.
+          if (
```

Comment:
The code below this doesn't seem like svgImage specific, If it's a duplicate code then can we find a way to move this to a service? (not sure, but let me know if that's possible!)

Reply:
This code is not very similar to the image directive because we don't need the reload and loading indicator here.